src/applications/disability-benefits/all-claims/validations.js

```diff
@@ -205,3 +206,19 @@ export const hasMonthYear = (err, fieldData) => {
     err.addError('Please provide both month and year');
   }
 };
+
+export const isWithinServicePeriod = (errors, fieldData, formData) => {
+  const inServicePeriod = _.get(
+    'serviceInformation.servicePeriods',
+    formData,
+    [],
+  ).some(pos => isWithinRange(fieldData, pos.dateRange));
+  if (!inServicePeriod) {
+    const dateIsComplete = dateString =>
+      dateString && !dateString.includes('X');
+    if (dateIsComplete(fieldData.from) && dateIsComplete(fieldData.to)) {
+      errors.from.addError('Must be within a single period of service.');
+      errors.to.addError('Must be within a single period of service.');
```

Comment:
🤔 yeah not ideal to have the error show in both spots as opposed to once at the top of the date range UI.
Wonder what we can do about this.

Reply:
Change the way validations work in USFS' `ObjectField`. :trollface: